src/include/wiredtiger.in

```diff
@@ -2837,10 +2837,12 @@ struct __wt_connection {
  * WT_CONNECTION::load_extension as the \c config parameter (for example\,
  * <code>extensions=(/path/ext.so={entry=my_entry})</code>)., a list of strings;
  * default empty.}
- * @config{file_extend, file extension configuration.  If set\, extend files of
- * the set type in allocations of the set size\, instead of a block at a time as
+ * @config{file_extend, file extension configuration.  If set for data files\,
+ * extend those in allocations of the set size\, instead of a block at a time as
  * each new block is written.  For example\,
- * <code>file_extend=(data=16MB)</code>., a list\, with values chosen from the
+ * <code>file_extend=(data=16MB)</code>. Disables log file extension when set to
+ * -1 for log file.  The default is to extend the log file in allocations of the
+ * <code>log_file_max</code> size., a list\, with values chosen from the
```

Comment:
I have a couple comments on making this configurable via `file_extend`:
1. It feels weird to make it specific only to log extension. I think someone may want to turn it off for either type of file.
2. Wouldn't setting it to 0 turn it off? How is 0 different than -1 here? What happens today if the user sets it to 0? I think they're the same, and if so, using 0 feels more natural.

Reply:
Ans 1. Currently, log files get extended irrespective of `file_extend` config.  See here:
https://github.com/wiredtiger/wiredtiger/blob/333460997573558235254cb7e84dad765bb844a7/src/log/log.c#L680]
So `file_extend` config is working differently for data files and log files, hence we are making the change specific only to log files.

Ans 2 : Setting it to 0 doesn't turn it off for log files. The default is 0, which means data file extension is off, but log file extension is on. So they are not the same. Hence to disable log file extension, set it to -1.